src/install-pnpm/run.ts

```diff
@@ -41,18 +41,32 @@ async function readTarget(opts: {
   readonly standalone: boolean
 }) {
   const { version, packageJsonFile, standalone } = opts
+  const { GITHUB_WORKSPACE } = process.env
 
-  if (version) return `${ standalone ? '@pnpm/exe' : 'pnpm' }@${version}`
+  let packageManager
+
+  if (GITHUB_WORKSPACE) {
+    ({ packageManager } = JSON.parse(await readFile(path.join(GITHUB_WORKSPACE, packageJsonFile), 'utf8')))
+  }
```

Comment:
If there are not `package.json` at root, this will result in error. Please fix this.

Reply:
Hm, this error was also there before, wasn't it? Should that be a separate PR?

Happy to wrap it in a `try/catch` to swallow the error if you think it belongs in this PR

Reply:
> Hm, this error was also there before, wasn't it? Should that be a separate PR?

Before, if user already specifies `version` in the action, no error would actually happen.

> Happy to wrap it in a `try/catch` to swallow the error if you think it belongs in this PR

If the file doesn't exist (`error.code === 'ENOENT'`), assign `undefined`.

Reply:
`packageManager` is already set to `undefined`

I added a condition to swallow the error if `error.code === 'ENOENT'`:

- b036c0dcb920715f3d914f0f8eafdcca573a9d7b